pkgs/applications/virtualization/crosvm/default.nix

```diff
@@ -0,0 +1,76 @@
+{ rustPlatform, stdenv, fetchgit, makeWrapper, minijail, binutils-unwrapped, pkgconfig, libusb1, buildPackages, libcap, runCommand, symlinkJoin }:
+
+let
+
+  arch = with stdenv.hostPlatform;
+    if isArm then "arm"
+    else if isx86_64 then "x86_64"
+    else throw "no seccomp policy files available for host platform";
+
+  # used to turn symlinks into real files because write permissions are necessary for the vendoring process
+  delink = src: runCommand src.name {
+    preferLocalBuild = true;
+    allowSubstitutes = false;
+  } ''
+    cp -prL --reflink=auto ${src} $out
+  '';
+
+  # used to place subtrees into the location they have in the Chromium monorepo
+  move = src: target: runCommand "moved-${src.name}" {
+    preferLocalBuild = true;
+    allowSubstitutes = false;
+  } ''
+    mkdir -p $(dirname $out/${target})
+    ln -s ${src} $out/${target}
+  '';
+
+  # used to check out subtrees from the Chromium monorepo
+  chromiumSource = name: subtrees: delink (symlinkJoin {
+    inherit name;
+    paths = stdenv.lib.mapAttrsToList (location: value:
+      move (fetchgit {
+        url = "https://chromium.googlesource.com/${location}.git";
+        inherit (value) rev sha256;
+      }) location) subtrees;
+  });
+
+in
+
+  rustPlatform.buildRustPackage rec {
+    pname = "crosvm";
+    version = "75";
+
+    src = chromiumSource "${pname}-sources" {
+      "chromiumos/platform/crosvm" = {
+        rev = "c23650c0ef8805d79beecc07ad1eada9aa3dda43";
+        sha256 = "1q9hcxc9vq2ih28zhq74z2sq4vg0h62alnpqfzh694gshzb1fxpf";
+      };
+      "chromiumos/third_party/adhd" = {
+        rev = "362b310772fb214a149e7ca5fa5268ec54ef01a5";
+        sha256 = "179bd2c3ax2gm08zg02dfzp2w5civ0ci7r5119pc7a69ads78nyb";
+      };
+    };
+
+    sourceRoot = "${pname}-sources/chromiumos/platform/crosvm";
+
+    patches = [
+      ./seccomp-policy-dir_env-var.patch
+    ];
+
+    cargoSha256 = "16cfp79c13ng5jjcrvz00h3cg7cc9ywhjiq02vsm757knn9jgr1v";
+
+    buildInputs = [ makeWrapper minijail binutils-unwrapped pkgconfig libusb1 (buildPackages.python2.withPackages (p: [ p.libfdt ])) libcap ];
+
+    postInstall = ''
+      mkdir -p $out/share/policy/
+      cp seccomp/${arch}/* $out/share/policy/
+      wrapProgram $out/bin/crosvm --set CROSVM_SECCOMP_POLICY_DIR $out/share/policy/
+    '';
+
+    meta = with stdenv.lib; {
+      description = "A secure virtual machine monitor for KVM";
+      homepage = https://chromium.googlesource.com/chromiumos/platform/crosvm/;
```

Comment:
Hint: We're probably going to deprecate unquoted URL syntax once RFC 45 is approved.

Reply:
Thanks, fixed.